x-pack/legacy/plugins/siem/public/components/matrix_histogram/index.test.tsx

```diff
@@ -52,36 +68,56 @@ describe('Matrix Histogram Component', () => {
     subtitle: 'mockSubtitle',
     totalCount: -1,
     title: 'mockTitle',
-    updateDateRange: jest.fn(),
+    dispatchSetAbsoluteRangeDatePicker: jest.fn(),
   };
-  describe('rendering', () => {
-    test('it renders EuiLoadingContent on initialLoad', () => {
-      const wrapper = shallow(<MatrixHistogram {...mockMatrixOverTimeHistogramProps} />);
 
-      expect(wrapper.find(`[data-test-subj="initialLoadingPanelMatrixOverTime"]`)).toBeTruthy();
+  beforeAll(() => {
+    (useQuery as jest.Mock).mockReturnValue({
+      data: null,
+      loading: false,
+      inspect: false,
+      totalCount: null,
     });
-
-    test('it renders Loader while fetching data if visited before', () => {
-      const mockProps = {
-        ...mockMatrixOverTimeHistogramProps,
-        data: [{ x: new Date('2019-09-16T02:20:00.000Z').valueOf(), y: 3787, g: 'config_change' }],
-        totalCount: 10,
-        loading: true,
-      };
-      const wrapper = shallow(<MatrixHistogram {...mockProps} />);
-      expect(wrapper.find('.loader')).toBeTruthy();
+    wrapper = mount(<MatrixHistogram {...mockMatrixOverTimeHistogramProps} />);
+  });
+  describe('on initial load', () => {
+    test('it renders MatrixLoader', () => {
+      expect(wrapper.html()).toMatchSnapshot();
+      expect(wrapper.find('MatrixLoader')).toHaveLength(1);
     });
+  });
 
-    test('it renders BarChart if data available', () => {
-      const mockProps = {
-        ...mockMatrixOverTimeHistogramProps,
-        data: [{ x: new Date('2019-09-16T02:20:00.000Z').valueOf(), y: 3787, g: 'config_change' }],
-        totalCount: 10,
+  describe('not initial load', () => {
+    beforeAll(() => {
+      (useQuery as jest.Mock).mockReturnValue({
+        data: [
+          { x: 1, y: 2, g: 'g1' },
+          { x: 2, y: 4, g: 'g1' },
+          { x: 3, y: 6, g: 'g1' },
+          { x: 1, y: 1, g: 'g2' },
+          { x: 2, y: 3, g: 'g2' },
+          { x: 3, y: 5, g: 'g2' },
+        ],
         loading: false,
-      };
-      const wrapper = shallow(<MatrixHistogram {...mockProps} />);
+        inspect: false,
+        totalCount: 1,
+      });
+      wrapper.setProps({ endDate: 100 });
+      wrapper.update();
+    });
+    test('it renders no MatrixLoader', () => {
+      expect(wrapper.html()).toMatchSnapshot();
+      expect(wrapper.find(`MatrixLoader`)).toHaveLength(0);
```

Comment:
For this and other instances of "the element does not exist"-style assertions, consider replacing the `.toHaveLength(0)` assertion with this form:

```
expect(wrapper.find(`MatrixLoader`).exists()).toBe(false);
```